batteries/result.luau

```diff
@@ -0,0 +1,26 @@
+type T<Value> =
+    | { success: true, value: Value }
+    | { success: false, traceback: string, err: string }
```

Comment:
stylua doesn't like this even though i think it's the nicest way to write this 😦